apisix/plugins/request-id.lua

```diff
@@ -41,9 +75,144 @@ function _M.check_schema(conf)
 end
 
 
+-- Generates the current process data machine
+local function gen_data_machine(max_number)
+    if data_machine == nil then
+        local etcd_cli, prefix = core.etcd.new()
+        local prefix = prefix .. "/plugins/request-id/snowflake/"
+        local uuid = uuid.generate_v4()
+        local id = 1
+        ::continue::
+        while (id <= max_number) do
+            local res, err = etcd_cli:grant(attr.snowflake.data_machine_ttl)
+            if err then
+                id = id + 1
+                core.log.error("Etcd grant failure, err: ".. err)
+                goto continue
+            end
+
+            local _, err1 = etcd_cli:setnx(prefix .. tostring(id), uuid)
```

Comment:
So we have to **write** ETCD so that we can write the uuid?

Reply:
The purpose of writing etcd here is to generate `DataMachineId` for snowflake. The value is uuid because `etcd_cli:setnx` still returns `succeeded` even if the key exists. UUID is used for verification after `etcd_cli:setnx`.